typings/index.d.ts

```diff
@@ -355,6 +406,28 @@ declare module 'discord.js' {
     public once(event: 'end', listener: (collected: Collection<K, V>, reason: string) => void): this;
   }
 
+  export class CommandInteraction extends Interaction {
+    public readonly command: ApplicationCommand | null;
+    public commandID: string;
+    public commandName: string;
+    public deferred: boolean;
+    public options: CommandInteractionOption[];
+    public replied: boolean;
+    public webhook: WebhookClient;
+    public defer(ephemeral?: boolean): Promise<void>;
+    public deleteReply(): Promise<void>;
+    public editReply(
+      content: string | APIMessage | WebhookEditMessageOptions | MessageEmbed | MessageEmbed[],
+      options?: WebhookEditMessageOptions,
+    ): Promise<Message | RawMessage>;
+    public fetchReply(): Promise<Message | RawMessage>;
+    public reply(
```

Comment:
should add some more overloads for this, as for instance `reply(MessageEmbed, InteractionReplyOptions)` or `reply(APIMessage, InteractionReplyoptions)` etc is not valid, which is accepted by the overload